backends/bmv2/backend.h

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2013-present Barefoot Networks, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef _BACKENDS_BMV2_BACKEND_H_
+#define _BACKENDS_BMV2_BACKEND_H_
+
+#include "ir/ir.h"
+#include "lib/error.h"
+#include "lib/exceptions.h"
+#include "lib/gc.h"
+#include "lib/json.h"
+#include "lib/log.h"
+#include "lib/nullstream.h"
+#include "analyzer.h"
+#include "helpers.h"
+#include "expressionConverter.h"
+#include "frontends/p4/coreLibrary.h"
+#include "midend/convertEnums.h"
+#include "frontends/common/model.h"
+#include "frontends/p4/fromv1.0/v1model.h"
+
+namespace BMV2 {
+
+class DirectMeterMap final {
```

Comment:
That does seem odd -- this class seems to be tracking the (color) output, but that should also be needed for meters and registers, which also have destination fields that are written by the extern.